### PR TITLE
JAVA-29305 Upgrade spring-security-web-boot-2

### PR DESCRIPTION
**Pull request by timis1**

### CODE REVIEW

File: spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multipleentrypoints/MultipleEntryPointsSecurityConfig.java [hunk 1]
```diff
@@ -90,11 +102,11 @@ public AuthenticationEntryPoint loginUrlauthenticationEntryPointWithWarning(){
...
-        public SecurityFilterChain filterChainApp3(HttpSecurity http) throws Exception {
-            http.antMatcher("/guest/**")
-                .authorizeRequests()
-                .anyRequest()
-                .permitAll();
+        public SecurityFilterChain filterChainApp3(HttpSecurity http, HandlerMappingIntrospector introspector) throws Exception {
+            MvcRequestMatcher.Builder mvcMatcherBuilder = new MvcRequestMatcher.Builder(introspector);
+            http.securityMatcher("/guest/**")
+                    .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry ->
+                            authorizationManagerRequestMatcherRegistry.requestMatchers(mvcMatcherBuilder.pattern("/guest/**")).permitAll());
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multipleentrypoints/MultipleEntryPointsSecurityConfig.java:109]: Please remove the extra empty line.

File: spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java [hunk 2]
```diff
@@ -44,30 +47,25 @@ public UserDetailsService userDetailsServiceApp1() {
...
-        public SecurityFilterChain filterChainApp1(HttpSecurity http) throws Exception {
-            http.antMatcher("/admin*")
-                .authorizeRequests()
-                .anyRequest()
-                .hasRole("ADMIN")
+        public SecurityFilterChain filterChainApp1(HttpSecurity http, HandlerMappingIntrospector introspector) throws Exception {
+            MvcRequestMatcher.Builder mvcMatcherBuilder = new MvcRequestMatcher.Builder(introspector);
+            http.securityMatcher("/admin*")
+                .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry ->
+                            authorizationManagerRequestMatcherRegistry.requestMatchers(mvcMatcherBuilder.pattern("/admin*")).hasRole("ADMIN"))
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java:54]: Please remove the extra empty line and take note of the line formatting.

File: spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java [hunk 3]
```diff
@@ -87,30 +85,25 @@ public UserDetailsService userDetailsServiceApp2() {
...
-        public SecurityFilterChain filterChainApp2(HttpSecurity http) throws Exception {
-            http.antMatcher("/user*")
-                .authorizeRequests()
-                .anyRequest()
-                .hasRole("USER")
+        public SecurityFilterChain filterChainApp2(HttpSecurity http, HandlerMappingIntrospector introspector) throws Exception {
+            MvcRequestMatcher.Builder mvcMatcherBuilder = new MvcRequestMatcher.Builder(introspector);
+            http.securityMatcher("/user*")
+                .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry ->
+                            authorizationManagerRequestMatcherRegistry.requestMatchers(mvcMatcherBuilder.pattern("/user*")).hasRole("USER"))
```
Comment (alvin577austria) [inline spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java:92]: Same

**Review by alvin577austria — CHANGES_REQUESTED** [review]
Few comments for minor fixes.

File: spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multipleentrypoints/MultipleEntryPointsSecurityConfig.java (hunk 1 above)
Comment (timis1) [reply spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multipleentrypoints/MultipleEntryPointsSecurityConfig.java]: Done

File: spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java (hunk 2 above)
Comment (timis1) [reply spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java]: Done

File: spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java (hunk 3 above)
Comment (timis1) [reply spring-security-modules/spring-security-web-boot-2/src/main/java/com/baeldung/multiplelogin/MultipleLoginSecurityConfig.java]: Done

**Pull request merged**